internal/processor/filtermatcher/attributematcher.go

```diff
@@ -105,8 +105,22 @@ func (ma attributesMatcher) Match(attrs pdata.AttributeMap) bool {
 				return false
 			}
 		} else if property.AttributeValue != nil {
-			if !attr.Equal(*property.AttributeValue) {
+			propertyAttrValue := property.AttributeValue
+			switch propertyAttrValue.Type() {
+			// Case when Attribute Values are of list values.
+			case pdata.AttributeValueARRAY:
+				for i := 0; i < propertyAttrValue.ArrayVal().Len(); i++ {
+					// Check attr value is exists in the AttributeValue array.
+					// Equal checks for any of int, string, bool and double.
+					if attr.Equal(propertyAttrValue.ArrayVal().At(i)) {
+						return true
+					}
+				}
 				return false
+			default:
+				if !attr.Equal(*property.AttributeValue) {
```

Comment:
handling at `attr.Equal(*property.AttributeValue)` would be ideal case. But `attr, exist := attrs.Get(property.Key)` returns `String` type for array values under `property.Key`. And further it always goes under string case execution

Reply:
Can you point to the code where the return value is string?

Reply:
Here [Get](https://github.com/open-telemetry/opentelemetry-collector/blob/1b61151c4a2ff9dc51716daaa39c882717edd02a/internal/processor/filtermatcher/attributematcher.go#L97) func. Though the `property.AttributeValue` is of ARRAY type (since `NewAttributeValueRaw()` returns `NewAttributeValueArray` as per my logic), the return value of Get method is string. 

Otherwise handling at [Equal](https://github.com/open-telemetry/opentelemetry-collector/blob/1b61151c4a2ff9dc51716daaa39c882717edd02a/consumer/pdata/common.go#L377) func would be ideal. In fact, there is a `// TODO: handle MAP and ARRAY data types.` I have tried adding these cases, but `attr` is returning  string and equal method is falling under string case. 

I have tried using `attr.SetArrayVal()` but this is changing whole attributes as array. 

Anything should i be considering for these scenarios?